Cauldron/Assets/Scripts/SO/MilestoneProvider.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+[CreateAssetMenu]
+public class MilestoneProvider : ScriptableObject
```

Comment:
Same as TutorialProvider, why is it scriptable if we don't need its inspector fields? Let's convert to regular class.